### PR TITLE
Unified structure

### PR DESCRIPTION
**Pull request by minijackson**
Made during the Documentathon 2023.

Merge the how-tos repository into this one, and unify structure by categories.

cc @gabrielfedel

### CODE REVIEW

**Comment by minijackson** [comment]
Converted to draft, because we still need to add URL redirects.

**Comment by minijackson** [comment]
So for some more description:

the appdevguide was left in its own directory, as-is (except for the epics build facility),
because I think the general direction we were going was to split the appdevguide bit by bit.
Each time the appdevguide is "split", the new page would be better integrated into the new structure.

A more complete description on how Diátaxis and this structure interact was written in the `CONTRIBUTING.md`. I think the the Diátaxis structure is great for writer, but not necessary for reader, so it's structured by theme, and then structured by Diátaxis classification.

Old page locations redirects to the new location, thanks to @stephane-cea.

Apart from fixing links, no content was changed.

**Comment by simon-ess** [comment]
Note that this PR adds some warnings:
```
/Users/simonrose/git/epics/docs/epics-docs/getting-started/linux-packages.rst:104: WARNING: Title underline too short.

Packages required to build MEDM
------------------------------
/Users/simonrose/git/epics/docs/epics-docs/getting-started/linux-packages.rst:104: WARNING: Title underline too short.

Packages required to build MEDM
------------------------------
/Users/simonrose/git/epics/docs/epics-docs/pv-access/Normative-Types-Specification.rst:365: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/simonrose/git/epics/docs/epics-docs/pv-access/Normative-Types-Specification.rst:421: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/simonrose/git/epics/docs/epics-docs/pv-access/Normative-Types-Specification.rst:443: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/simonrose/git/epics/docs/epics-docs/pv-access/Normative-Types-Specification.rst:2404: WARNING: Content block expected for the "table" directive; none found.

.. table:: Type Names that may be used in the Type Name part of a
/Users/simonrose/git/epics/docs/epics-docs/pv-access/Normative-Types-Specification.rst:2405: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/simonrose/git/epics/docs/epics-docs/pv-access/Normative-Types-Specification.rst:2464: WARNING: Literal block expected; none found.
```

**Comment by minijackson** [comment]
@simon-ess all warnings should be fixed

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -47,19 +47,35 @@ a developer should pick it up.
...
+The various documents are divided by topic,
+for example:
+
+-   Getting started
+-   Process database
+-   Modules
+-   etc.
+
+This documentation also follows the [Diátaxis] documentation framework.
 We recommend reading the [Diátaxis] documentation.
```
Comment (simon-ess) [inline CONTRIBUTING.md:59]: Very minor (and not from this change), it seems odd to have the same link twice in back-to-back sentences.

**Comment by simon-ess** [comment]
Perhaps this is out-of-scope, but it seems that this change massively changes the amount of headings in the side-bar. Compare https://epics--18.org.readthedocs.build/en/18/index.html to https://docs.epics-controls.org/en/latest/; you have to scroll down quite far on the current PR!

**Comment by simon-ess** [comment]
A few other comments: it is a bit odd at the moment that we have some top-level links that point to external sites (for example, all of the modules pages are links to external sites, but so is the link to the PV access specification). Note that this will be fixed in a future PR

**Comment by minijackson** [comment]
@simon-ess approved verbally. Merging it since other PRs depends on it.

**Pull request merged**